mongo/document/document.go

```diff
@@ -0,0 +1,144 @@
+package document
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+
+	"github.com/Scalingo/go-utils/logger"
+	"github.com/Scalingo/go-utils/mongo"
+)
+
+type Document interface {
+	GetID() bson.ObjectId
+}
+
+type ParanoiaDeletable interface {
+	Updatable
+	SetDeletedAt(time.Time)
+}
+
+type Updatable interface {
+	Document
+	SetUpdatedAt(time.Time)
+}
+
+func Save(ctx context.Context, collectionName string, doc Document) error {
+	log := logger.Get(ctx)
+	c := mongo.Session(log).Clone().DB("").C(collectionName)
+	defer c.Database.Session.Close()
+	log.WithField(collectionName, doc).Debugf("save '%v'", collectionName)
+	_, err := c.UpsertId(doc.GetID(), &doc)
+	return err
+}
+
+// Remove the volume from the database.
+// Handle with care...
+func Destroy(ctx context.Context, collectionName string, doc Document) error {
+	log := logger.Get(ctx)
+	c := mongo.Session(log).Clone().DB("").C(collectionName)
+	defer c.Database.Session.Close()
+	log.WithField(collectionName, doc).Debugf("remove '%v'", collectionName)
+	return c.RemoveId(doc.GetID())
```

Comment:
Mask the error

Reply:
As it's the only source of error, I actually don't think it's require to mask the error here, with the stacktrace we'll definitely find the source of the error.